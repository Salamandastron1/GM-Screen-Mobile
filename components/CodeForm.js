import React, { Component } from 'react';
import { Alert } from 'react-native';
import Title from './Title';
import MainContain from './MainContain';
import CardSection from './CardSection';
import Input from './Input';
import Button from './Button';
import apiCall from '../helpers/API';

class CodeForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      code: '',
    };

    this.handleInput = this.handleInput.bind(this);
  }

  handleInput({ nativeEvent }) {
    this.setState({
      code: nativeEvent.text,
    });
  }

  async determineRoute(code) {
    const { navigation: navigate } = this.props;
    const url = `/api/v1/characters/play_code/${code}`;
    const playerInfo = await apiCall(url);

    if (playerInfo.created === 1) {
      navigate('PlayerScreen');
    } else if (!playerInfo.created === 0) {
      navigate('CharacterForm');
    } else {
      Alert.alert('go fuck yourself');
    }
  }

  render() {
    const { code } = this.state;

    return (
      <MainContain>
        <CardSection>
          <Input
            value={code}
            onChange={this.handleInput}
            name="code"
            length={6}
            placeholder="Enter Code"
          />
        </CardSection>
        <CardSection>
          <Button
            onPress={() => this.determineRoute(code)}
          >
            Submit
          </Button>
        </CardSection>
        <Title>
          Enter Code
        </Title>
      </MainContain>
    );
  }
}

export default CodeForm;
