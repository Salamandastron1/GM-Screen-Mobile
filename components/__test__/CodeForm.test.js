/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import CodeForm from '../CodeForm';

it('should match the shapshot', () => {
  const tree = shallow(<CodeForm />);
  expect(tree).toMatchSnapshot();
});