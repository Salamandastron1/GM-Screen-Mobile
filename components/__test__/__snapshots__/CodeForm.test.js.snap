// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the shapshot 1`] = `
<MainContain>
  <CardSection>
    <Input
      length={6}
      name="code"
      onChange={[Function]}
      placeholder="Enter Code"
      value=""
    />
  </CardSection>
  <CardSection>
    <Button
      onPress={[Function]}
    >
      Submit
    </Button>
  </CardSection>
  <Title>
    Enter Code
  </Title>
</MainContain>
`;
